examples/connext_dds/builtin_topics/c++11/msg_publisher.cxx

```diff
@@ -145,50 +152,55 @@ void publisher_main(int domain_id, int sample_count)
     // participant_qos << resource_limits_qos;
 
     // Create a DomainParticipant with default Qos.
-    DomainParticipant participant(domain_id);
+    dds::domain::DomainParticipant participant(domain_id);
 
     // Installing listeners for the builtin topics requires several steps
     // First get the builtin subscriber.
-    Subscriber builtin_subscriber = dds::sub::builtin_subscriber(participant);
+    dds::sub::Subscriber builtin_subscriber =
+            dds::sub::builtin_subscriber(participant);
+
+    // Create shared pointer to BuiltinParticipantListener class
+    std::shared_ptr<BuiltinParticipantListener> participant_listener =
```

Comment:
```suggestion
    auto participant_listener =
```

Use `auto`; the result of `make_shared` is well-known, so declaring the type is redundant and less readable.